### PR TITLE
feat: split CRD group into cluster and placement

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
### Description of your changes
Split the v1beta1 CRD group into cluster and placement.

Fixes #

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### Special notes for your reviewer
All changes are just find and replace.

### CODE REVIEW

File: apis/cluster/v1beta1/doc.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package v1beta1 contains API Schema definitions for the fleet cluster v1beta1 API group
```
Comment (zhiying-lin) [inline apis/cluster/v1beta1/doc.go:6]: missing period

File: apis/cluster/v1beta1/groupversion_info.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// +groupName=cluster.kubernetes-fleet.io
+package v1beta1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+var (
+	// GroupVersion is group version used to register these objects
```
Comment (zhiying-lin) [inline apis/cluster/v1beta1/groupversion_info.go:16]: missing period

File: pkg/controllers/internalmembercluster/v1beta1/member_controller_test.go [hunk 3]
```diff
@@ -8,6 +8,7 @@ package v1beta1
 import (
 	"context"
 	"errors"
+
```
Comment (zhiying-lin) [inline pkg/controllers/internalmembercluster/v1beta1/member_controller_test.go:11]: remove the line?

File: pkg/scheduler/framework/plugins/sameplacementaffinity/filtering.go [hunk 4]
```diff
@@ -8,16 +8,18 @@ package sameplacementaffinity
 import (
 	"context"
 
-	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+
```
Comment (zhiying-lin) [inline pkg/scheduler/framework/plugins/sameplacementaffinity/filtering.go:13]: redundant line

File: apis/placement/v1beta1/commons.go [hunk 5]
```diff
@@ -5,13 +5,6 @@ Licensed under the MIT license.
...
-import (
-	v1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-)
-
-type ClusterState string
-
 const (
 	// Unprefixed labels/annotations are reserved for end-users
 	// we will add a kubernetes-fleet.io to designate these labels/annotations as official fleet labels/annotations.
```
Comment (zhiying-lin) [inline apis/placement/v1beta1/commons.go:10]: update the fleetPrefix as "placement.kubernetes-fleet.io/"?

**Review by zhiying-lin — APPROVED** [review]
Few minor comments, approve the PR to unblock others

File: apis/placement/v1beta1/commons.go (hunk 5 above)
Comment (ryanzhang-oss) [reply apis/placement/v1beta1/commons.go:10]: I was thinking this is the common prefix for all.

Comment (zhiying-lin) [reply apis/placement/v1beta1/commons.go:10]: that works too, need to move this common prefix out of the apis/placement/beta1/common.go to  a common place, eg, apis/common.go?

**Pull request merged**